Add plugin: Discord Message Sender

okawak opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/okawak/discord_message_sender

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

mnaoumov commented:
## 1

Released plugin cannot be loaded

```
app.js:1 Plugin failure: discord-message-sender Error: WASM initialization failed: Error: ENOENT: no such file or directory, open 'path\to\vault\.obsidian\plugins\discord-message-sender\parse_message_bg.wasm'
    at eval (VM393 plugin:discord-message-sender:2:660562)
    at async RA.onload (VM393 plugin:discord-message-sender:2:661033)
```

## 2

1. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/styles.css#L1

Remove the `styles.css` file from your release, since it's empty it's just unnecessary bloat that Obsidian would download.

## 3

Missing [`versions.json`](https://docs.obsidian.md/Reference/Versions) file in the repo root.

## 4

1. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/discordApi.ts#L70
2. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/main.ts#L24
3. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/main.ts#L42
4. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/main.ts#L60
5. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/settingTab.ts#L30
6. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/settingTab.ts#L30
7. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/settingTab.ts#L40
8. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/settingTab.ts#L55
9. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/settingTab.ts#L75

[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

## 5

1. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/utils.ts#L2

Use [`sleep()`](https://github.com/obsidianmd/obsidian-api/blob/9ab497e/obsidian.d.ts#L252) instead.

## 6

1. https://github.com/okawak/discord_message_sender/blob/1de2c276eb91251b1530faf1b9ee9e1425edbad9/src/wasmBridge.ts#L22

Useless check, `adapter` is always of type `DataAdapter`

okawak commented:
@mnaoumov

Thank you very much for your detailed code review. I’ve addressed the issues you pointed out in items 3, 4, 5, and 6, and have released an updated version as 0.1.10.

Regarding point 1, I believe the `.wasm` file is included in the release. Since the functionality depends on this file, the code won’t work without it. Is including it in the release not sufficient?

As for point 2, the style.css file wasn’t included in the release archive for version 0.1.9, but to avoid any confusion, I’ve removed it from the repository as well.

I’d appreciate it if you could take another look when you have time.
Thanks again!

okawak commented:
@mnaoumov 

Thank you for your comment. I've updated the code to inline the `.wasm` file into the `main.js`, and released version 0.2.0. I’d appreciate it if you could take a look.

joethei commented:
[discord](https://github.com/okawak/discord_message_sender/blob/8cf6baf0c6942569a562e4baac79a10f7b135c28/src/main.ts#L41)
Discord is a brand name and is capitalized.

[if (vault.getAbstractFileByPath(p)) return;](https://github.com/okawak/discord_message_sender/blob/8cf6baf0c6942569a562e4baac79a10f7b135c28/src/vault.ts#L24)
This can fail if there is a file with no extension using the same name.
Do a `instanceof TFolder` check.

okawak commented:
@joethei 

Thank you for the code review!
Based on your comments, I’ve made the changes and released the updated version as 0.2.2.

I’d appreciate it if you could take another look.
